### DISCUSSION TITLE
Nested tabbed pages and their own "go scope"?

### DISCUSSION THREAD

**Discussion by leoshusar** [csells/go_router#245] [Q&A]
Hi! I am creating multiple nested tabbed pages like this
![image](https://user-images.githubusercontent.com/13891066/146422658-898acdbf-910d-4e12-b7f6-75e75ac297a8.png)

What I want to do:
- from main page you open user info
- - on user info you can switch between user's roles, permissions...
- - here you can open some role detail
- - - on role detail you can switch between role's permissions, users assigned to this role...
- - - here you can open open some another user info

basically you can go in cycle like user info -> role info -> user info -> role info...

Now the problem. I want, when you press the back button, go back one page in the stack. Even if you change tab on that page, I don't want to go back to previous tab, but the whole page.
You're here:
`main -> user info -> role info -> user info -> role info`
Back:
`main -> user info -> role info -> user info`
Back:
`main -> user info -> role info`
and so on.

This doesn't work since when you use `push` to navigate to a new page and then `go` to navigate between tabs on that page, it clears the stack - because that's what `go` does.
And when I tried to use `push` for navigating between tabs too, I got an exception (is this bug or expected behavior?):

```
════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown building InheritedGoRouter(goRouter: Instance of 'GoRouter'):
Assertion failed:
..\…\widgets\navigator.dart:3470
!keyReservation.contains(key)
is not true
```

but I guess even if it worked, it would add each tab change to the stack, right? And the back button would just switch to the previous selected tab?

So, is there any way to achieve this? If not, I've got an idea for a feature request how this *might* be done.

**Comment by csells** [comment]
Not yet but it's definitely on the list for the future (although we're not quite sure of the right way to do it yet: https://github.com/csells/go_router/discussions/222).
